config/src/main/java/com/typesafe/config/impl/SerializedConfigValue.java

```diff
@@ -311,7 +324,21 @@ private static void writeValueData(DataOutput out, ConfigValue value) throws IOE
             out.writeUTF(((ConfigNumber) value).transformToString());
             break;
         case STRING:
-            out.writeUTF(((ConfigString) value).unwrapped());
+            String strVal = ((ConfigString) value).unwrapped();
+            if (strVal.length() > MAX_BYTES_LENGTH)
+                longStr = true;
+            List<String> values = new ArrayList<>();
+            if (strVal.getBytes().length >= MAX_BYTES_LENGTH) {
+                    values.addAll(splitt(strVal, 32767));
```

Comment:
should 32767 be MAX_BYTES_LENGTH/2?

Reply:
Yes, I guess it is. As I mentioned in my very first comment, I referred to https://github.com/geotools/geotools/commit/7b52e15943846f72f0d85cc9121d330daf8c6c6e to make a fixation. We are doing a research about automatically recommending related fixations for developers to help them figure out possible solutions more quickly. https://github.com/geotools/geotools/commit/7b52e15943846f72f0d85cc9121d330daf8c6c6e is the recommended one for solving this bug. I manually checked if the fix pattern in https://github.com/geotools/geotools/commit/7b52e15943846f72f0d85cc9121d330daf8c6c6e applies to this bug to evaluate our recommendation system. Thanks for your reply.